src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -348,6 +362,20 @@ private static (MethodInfo, Type[]) GetCachedMethodInfo(AssemblyKey assemblyKey,
         }
     }
 
+    [UnconditionalSuppressMessage(
+        "ReflectionAnalysis",
+        "IL2060:MakeGenericMethod",
+        Justification = "https://github.com/mono/linker/issues/1727")]
+    private static Task GetTaskByType(Type type, object obj)
+    {
+        var assemblyMethods = _cachedConvertToTaskByType.GetOrAdd(type, (Type t)=>
+            _taskConverterMethodInfo.MakeGenericMethod(t).CreateDelegate<Func<object, Task>>());
+
+        return assemblyMethods.Invoke(obj);
```

Comment:
The name `assemblyMethods` doesn't seem right. Is this deliberate? Suggestion:

```suggestion
        var converterDelegate = _cachedConvertToTaskByType.GetOrAdd(type, (Type t) =>
            _taskConverterMethodInfo.MakeGenericMethod(t).CreateDelegate<Func<object, Task>>());

        return converterDelegate.Invoke(obj);
```